mlflow/store/tracking/rest_store.py

```diff
@@ -54,16 +55,28 @@ def _call_endpoint(self, api, json_body):
         response_proto = api.Response()
         return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
 
-    def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
+    def list_experiments(
+        self, view_type=ViewType.ACTIVE_ONLY, max_results=None, page_token=None,
+    ):
         """
-        :return: a list of all known Experiment objects
+        :param view_type: Qualify requested type of experiments.
+        :param max_results: If passed, specifies the maximum number of experiments desired. If not
+                            passed, the server will pick a maximum number of results to return.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                            a ``list_experiments`` call.
+        :return: a PagedList of all known Experiment objects
```

Comment:
Awesome! @harupy can we also updated the return type docstring for `search_runs` to reflect that a PagedList is returned? The current return type is technically inaccurate: https://github.com/mlflow/mlflow/blob/1c43176cefb5531fbb243975b9c8c5bfb9775e66/mlflow/tracking/_tracking_service/client.py#L378-L380